src/main/java/house/inksoftware/systemtest/domain/steps/response/rest/ExpectedRestResponseStep.java

```diff
@@ -1,29 +1,251 @@
 import house.inksoftware.systemtest.domain.utils.JsonUtils;
 import lombok.Data;
 import lombok.SneakyThrows;
+
+import org.jetbrains.annotations.NotNull;
 import org.json.JSONException;
 import org.junit.Assert;
 
+import java.util.*;
+import java.util.stream.IntStream;
+
+import okhttp3.*;
+
+import static org.junit.Assert.*;
+
 @Data
 public class ExpectedRestResponseStep implements ExpectedResponseStep {
     private final int httpCode;
-    private final String body;
+    private final JsonNode body;
+    private final JsonNode verification;
+    private final Set<String> verificationFieldsSet;
+    private final Set<String> allFieldsSet;
 
+    @SneakyThrows
     public static ExpectedRestResponseStep from(String json) {
-        int httpCode = JsonPath.parse(json).read("httpCode");
-        String body = JsonPath.parse((Object) JsonPath.parse(json).read("body")).jsonString();
+        var objectMapper = new ObjectMapper();
+        var root = objectMapper.readTree(json);
+
+        var httpCode = root.get("httpCode").asInt();
+        var body = root.path("body");
+        var verification = root.path("verification");
 
-        return new ExpectedRestResponseStep(httpCode, body);
+        Set<String> verificationFieldSet = new HashSet<>();
+        if (!verification.isMissingNode()) {
+            var iterator = verification.elements();
+            iterator.forEachRemaining(field -> verificationFieldSet.add(field.get("field").asText()));
+        }
+        Set<String> allFieldSet = new HashSet<>();
+        allFieldSet = findAllFieldNamesFromJSON(allFieldSet, "", body);
+        return new ExpectedRestResponseStep(httpCode, body, verification, verificationFieldSet, allFieldSet);
     }
 
     @Override
-    public void assertResponseIsCorrect(ActualResponse response) throws JSONException {
-        JsonUtils.assertJsonEquals(body, response.body());
+    @SneakyThrows
+    public void assertResponseIsCorrect(ActualResponse response) {
+        if (verification.isMissingNode()) {
+            compareIfExactSame(response);
+            return;
+        }
+        var objectMapper = new ObjectMapper();
+        var actualResponseBody = objectMapper.readTree(response.body());
+
+        compareHttpStatuses(actualResponseBody);
+
+        actualResponseBody = actualResponseBody.path("body");
+
+        compareUsingVerification(actualResponseBody);
+        compareNormalWay(actualResponseBody);
+
+    }
+
+    private void compareHttpStatuses(JsonNode actualResponseBody) {
+        assertEquals(httpCode, actualResponseBody.get("httpCode").asInt());
+    }
+
+    private void compareNormalWay(JsonNode actualResponseBody) {
+        var fieldsToBeComparedNormally = findAllFieldsToBeComparedNormally();
+
+        fieldsToBeComparedNormally
+                .forEach(field -> compareFieldNormally(actualResponseBody, field));
+    }
+
+    private void compareFieldNormally(JsonNode actualResponseBody, String field) {
```

Comment:
Same here and everywhere above, it needs to be explicit, normally could mean lots of things